### PR TITLE
fix: handled to hide widget in preview mode

### LINKED ISSUE #10228: [Bug]: App in preview mode shows hidden widgets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hidden widgets should not be visible in **preview mode**

### Steps To Reproduce

1. Turn off `Visible` property of a widget and switch to **preview mode**. Observe hidden widget is still visible

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

>  Add Wrapper component for render preview mode. this component is responsible for hiding and showing content 
>   components base on isVisible 

Fixes #10228 

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/preview-mode-hide-widget 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.71 **(0.01)** | 37.06 **(0)** | 35.75 **(0.01)** | 56.07 **(0.01)**
 :sparkles: :new: | **app/client/src/components/editorComponents/PreviewModeComponent.tsx** | **90** | **50** | **100** | **88.89**
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 88.98 **(-0.78)** | 70.59 **(-1.96)** | 100 **(0)** | 92.38 **(-0.95)**
 :green_circle: | app/client/src/widgets/BaseWidget.tsx | 79.53 **(0.67)** | 44.44 **(0)** | 75.76 **(0.76)** | 78.69 **(0.72)**</details>

### CODE REVIEW

File: app/client/src/components/editorComponents/PreviewModeComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+/**
+ * render only visible components in preview mode
+ */
+function PreviewModeComponent({ children, isVisible }: Props) {
+  const isPreviewMode = useSelector(previewModeSelector);
+
+  if (isPreviewMode && !isVisible) {
+    return null as any;
```
Comment (sbalaji1192) [inline app/client/src/components/editorComponents/PreviewModeComponent.tsx:17]: do we need `as any` ?

**Review by sbalaji1192 — COMMENTED** [review]
@riodeuno Could you please have a look at these changes?

File: app/client/src/components/editorComponents/PreviewModeComponent.tsx (hunk 1 above)
Comment (riodeuno) [reply app/client/src/components/editorComponents/PreviewModeComponent.tsx:17]: @techbhavin I agree with @sbalaji1192 we don't need to return anything here if the condition is false, I believe.

We can simply say `if(!(isPreviewMode && !isVisible)) return children;`. Please see if this approach works.

**Review by riodeuno — CHANGES_REQUESTED** [review]
This needs a proper cypress test for the following:
- A successful case, where an invisible widget in preview mode doesn't show up
- An unsuccessful case, where a visible widget in preview mode shows up (We probably have this one)

File: app/client/src/components/editorComponents/PreviewModeComponent.tsx (hunk 1 above)
Comment (sbalaji1192) [reply app/client/src/components/editorComponents/PreviewModeComponent.tsx:17]: we can use a much simpler condition `if (!isPreviewMode || isVisible) return children;`

**Comment by sbalaji1192** [comment]
> This needs a proper cypress test for the following:
> 
> * A successful case, where an invisible widget in preview mode doesn't show up
> * An unsuccessful case, where a visible widget in preview mode shows up (We probably have this one)

And a case to make sure that the invisible widgets show up in the edit mode.

File: app/client/src/components/editorComponents/PreviewModeComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  isVisible?: boolean;
+};
+
+/**
+ * render only visible components in preview mode
+ */
+function PreviewModeComponent({ children, isVisible }: Props) {
+  const isPreviewMode = useSelector(previewModeSelector);
+  if (!(isPreviewMode && !isVisible)) return children;
+  return null as any;
```
Comment (riodeuno) [inline app/client/src/components/editorComponents/PreviewModeComponent.tsx:16]: @techbhavin I am curious as to why would we need `any` here? Can we try returning a `ReactNode` or something like that? Can we return a Fragment?

**Review by riodeuno — APPROVED** [review]
Looks good to me! @techbhavin let's move this to the QA pipeline.

**Comment by shwetha-ramesh** [comment]
@techbhavin tested this PR. DP works fine. Widgets when ```Visible``` is off doesn't show up in Preview mode

image

Moving to Done.

**Pull request merged**